src/Microsoft.ML.Transforms/PermutationFeatureImportanceRegression.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Float = System.Single;
+
+namespace Microsoft.ML.Transforms
+{
+    using Stopwatch = System.Diagnostics.Stopwatch;
+
+    public sealed class PermutationFeatureImportanceRegression
+    {
+        private readonly MLContext _mlContext;
+        private VBuffer<ReadOnlyMemory<char>> _slotNames;
+        private readonly List<(string featureName, RegressionEvaluator.Result metricsDelta)> _metricsStats;
+
+        public PermutationFeatureImportanceRegression(MLContext mlContext)
+        {
+            Contracts.CheckValue(mlContext, nameof(mlContext));
+            _mlContext = mlContext;
+            _metricsStats = new List<(string featureName, RegressionEvaluator.Result metricsDelta)>();
+        }
+
+        public List<(string featureName, RegressionEvaluator.Result metricsDelta)>
+            GetImportanceMetricsMatrix(ITransformer model, IDataView data,
+            string label = DefaultColumnNames.Label, string features = DefaultColumnNames.Features,
+            int topExamples = 0, int progressIterations = 10)
+        {
+            var host = ((IHostEnvironment)_mlContext).Register(nameof(GetImportanceMetricsMatrix));
```

Comment:
>var host = ((IHostEnvironment)_mlContext).Register(nameof(GetImportanceMetricsMatrix)); [](start = 12, length = 87)

Can you move host creation to constructor of class rather than it's method?